### ISSUE TITLE
This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export

### ISSUE THREAD

**Issue by amitlevy21** [supercharge/promise-pool#38]
Hi, 

I want to remove the `esModuleInterop` flag from my projects `tsconfig.json`.

However, I'm using this awesome library and It's the only dependency I can't seem to find a way around it.

Is there a way to use promise-pool without having `esModuleInterop` flag?

tried:
```Typescript
import PromisePool from '@supercharge/promise-pool'

import * as PromisePool from '@supercharge/promise-pool'

import { withConcurrency } from '@supercharge/promise-pool'
```

I Get the compilation error with each

Thank you!

**Comment by marcuspoehls** [comment]
@amitlevy21 Hey Amit, thank you for reporting this.

I can reproduce your issue in a simple TypeScript project that doesn’t use the `esModuleInterop` flag. Looks like the package exports from `@supercharge/promise-pool` don’t support proper imports in TypeScript. I’ll think about a solution and probably fix it later today.

I appreciate a pull request if you know a way to resolve this issue. My feeling here is we need to change the package exports to explicitly export a class allowing you to create a promise pool.

**Comment by amitlevy21** [comment]
@marcuspoehls Hey Marcus, appreciate the quick response!

I'd love to help, from my understanding, we just need to change https://github.com/supercharge/promise-pool/blob/main/src/index.ts#L5

I'll take a look soon and share a PR

**Comment by marcuspoehls** [comment]
@amitlevy21 Thank you!

**Comment by marcuspoehls** [comment]
I just published version `2.0.0` to NPM 😃 

Thank you again @amitlevy21 for your help fixing this TypeScript importing issue when not using the `esModuleInterop` flag 🤝
